tests/Unit/Evolution/Particles/MonteCarlo/Test_EmitPackets.cpp

```diff
@@ -0,0 +1,91 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Particles/MonteCarlo/Packet.hpp"
+#include "Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Framework/TestingFramework.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+
+SPECTRE_TEST_CASE("Unit.Evolution.Particles.MonteCarloEmission",
+                  "[Unit][Evolution]") {
+  using Particles::MonteCarlo::Packet;
+
+  // Vector of MC packets
+  std::vector<Packet> all_packets = {};
+
+  const double time = 0.0;
+  const double time_step = 0.1;
+
+  MAKE_GENERATOR(generator);
+  const Mesh<3> mesh(3, Spectral::Basis::FiniteDifference,
+                     Spectral::Quadrature::CellCentered);
+
+  // Zero vector for tensor creations
+  DataVector zero_dv(27);
+  zero_dv = 0.0;
+
+  std::array<std::array<DataVector, 2>, 2> emission_in_cells = {
+      std::array<DataVector, 2>{{zero_dv, zero_dv}},
+      std::array<DataVector, 2>{{zero_dv, zero_dv}}};
+  std::array<DataVector, 2> single_packet_energy = {zero_dv, zero_dv};
+  const std::array<double, 2> energy_at_bin_center = {2.0, 5.0};
+  InverseJacobian<DataVector, 4, Frame::Inertial, Frame::Fluid>
+      inverse_jacobian = make_with_value<
+          InverseJacobian<DataVector, 4, Frame::Inertial, Frame::Fluid>>(
+          zero_dv, 0.0);
+  Jacobian<DataVector, 4, Frame::Inertial, Frame::Fluid> jacobian =
+      make_with_value<Jacobian<DataVector, 4, Frame::Inertial, Frame::Fluid>>(
+          zero_dv, 0.0);
+
+  // Set data
+  for (size_t s = 0; s < 2; s++) {
+    for (size_t i = 0; i < 27; i++) {
```

Comment:
`27` -> `single_packet_energy[0].size()` just in case the size gets adjusted at some point